### DISCUSSION TITLE
Not Listening on Local/LAN Interface

### DISCUSSION THREAD

**Discussion by bigverm23** [AdguardTeam/AdGuardHome#3594] [Q&A]
### Issue Details

AdguardHome Docker is not listening on the local docker host interface

Docker version 19.03.8, build afacb8b7f0
2021/09/13 16:19:37.547204 [info] AdGuard Home, version v0.106.3
2021/09/13 16:19:37.548650 [info] Go to http://127.0.0.1:3000
2021/09/13 16:19:37.548658 [info] Go to http://172.21.0.13:3000

### Expected Behavior
Run docker container and be directed to http://dockerhostIP:3000 for setup

### Actual Behavior
Run docker container and directed to http://172.21.0.13:3000 for setup
 
![Screenshot from 2021-09-13 12-28-57](https://user-images.githubusercontent.com/17554438/133121870-795606d8-fdda-4326-b33d-8ab2ee6fc272.png)

Docker-Compose: https://pastebin.com/iNdG99R0

**Comment by ameshkov** [comment]
I am not sure what you mean, AGH listens on all available network interfaces on the screenshot.

Note, that from inside of a docker container we cannot listen to your host's network interfaces, that's what port forwarding is for.

**Reply by bigverm23** [reply]
this is what I expect it to do, which is what happens when I install on the host machine using curl (non-Docker) 

021/09/14 11:17:04 [info] AdGuard Home is available on the following addresses:
2021/09/14 11:17:04 [info] Go to http://127.0.0.1:3000
2021/09/14 11:17:04 [info] Go to http://[::1]:3000
2021/09/14 11:17:04 [info] Go to http://10.0.0.12:3000
2021/09/14 11:17:04 [info] Go to http://172.21.0.1:3000
2021/09/14 11:17:04 [info] Go to http://172.24.0.1:3000
2021/09/14 11:17:04 [info] Go to http://172.17.0.1:3000

**Comment by bigverm23** [comment]
that is not entirely true.  Like i mentioned, my other containers running services like Radarr, Sonarr, Plex are all available using my host IP, however, Adguard is not.  It only sees the loopback interface and the docker container IP (not host) which causes an issue as it doesnt allow Adguard to differentiate clients on the LAN.  Adguard thinks all clients are running through the container IP, which again is not the case.

**Comment by ainar-g** [comment]
Your compose file has `3030:3000/tcp`, so the address should be `http://dockerhostIP:3030`, no?  Or am I missing something?

**Reply by bigverm23** [reply]
yes you are correct, except when I enter dockerhostIP:3030 , it does not resolve.  My expectation is that running the docker container would also show my LAN network interface, which it does not.

**Reply by bigverm23** [reply]
this is what I expect it to do, which is what happens when I install on the host machine using curl (non-Docker) 

021/09/14 11:17:04 [info] AdGuard Home is available on the following addresses:
2021/09/14 11:17:04 [info] Go to http://127.0.0.1:3000
2021/09/14 11:17:04 [info] Go to http://[::1]:3000
2021/09/14 11:17:04 [info] Go to http://10.0.0.12:3000
2021/09/14 11:17:04 [info] Go to http://172.21.0.1:3000
2021/09/14 11:17:04 [info] Go to http://172.24.0.1:3000
2021/09/14 11:17:04 [info] Go to http://172.17.0.1:3000

**Comment by samstarnes** [comment]
No, you're not missing anything, @ainar-g. I'm getting the same error that @bigverm23 is getting too.

Running through docker but not docker compose.

```
sudo docker run --name adguardhome \
    --restart unless-stopped \
    -v /srv/docker/adguardhome/data:/opt/adguardhome/work \
    -v /srv/docker/adguardhome/conf:/opt/adguardhome/conf \
    -p 53:53/tcp -p 53:53/udp \
    -p 8660:80/tcp -p 8661:443/tcp -p 8661:443/udp -p 8662:3000/tcp \
    -p 853:853/tcp -p 853:853/udp \
    -p 784:784/udp  -p 8853:8853/udp \
    -p 5443:5443/tcp -p 5443:5443/udp \
    -d adguard/adguardhome
```

Replacing ports 80 and 443 since they're already in use and 3000 as well since many developers use that and is often used. I get the same problem with the dashboard not forwarding correctly because it only identifies two interfaces. Docker, `172.17.0.9`, and host, `127.0.0.1`.
I should be expecting to be forwarded to `192.168.50.138:8662` but it fails to load. Everything else like gitea, netdata, portainer, etc can all be accessed through `192.168.50.138:port` but not AGH.

`Docker version 20.10.8, build 3967b7d28e` 
Latest AdGuardHome.

**Comment by samstarnes** [comment]
Hmmm, well I decided to run it as `--network host` instead.

```
sudo docker run --name adguardhome \
    --restart unless-stopped \
    -v /srv/docker/adguardhome/data:/opt/adguardhome/work \
    -v /srv/docker/adguardhome/conf:/opt/adguardhome/conf \
    --network host \
    -d adguard/adguardhome
```

It gives me *every* interface as an option. I chose `192.168.50.138` and switched the admin interface port to something different. Works as intended but port 80/443 is still being used and the dashboard is still using 3000.

```
CONTAINER ID   IMAGE                                      COMMAND                  CREATED          STATUS                       PORTS                                                                                      NAMES
c0499757a69d   adguard/adguardhome                        "/opt/adguardhome/Ad…"   43 minutes ago   Up 43 minutes                                                                                                           adguardhome


2021/09/20 02:21:42.075301 [info] This is the first launch of AdGuard Home, redirecting everything to /install.html 
2021/09/20 02:21:42.075339 [info] AdGuard Home is available on the following addresses:
2021/09/20 02:21:42.079501 [info] Go to http://127.0.0.1:3000
2021/09/20 02:21:42.079509 [info] Go to http://[::1]:3000
2021/09/20 02:21:42.079530 [info] Go to http://192.168.50.138:3000
2021/09/20 02:21:42.079532 [info] Go to http://[2601:c6:d680:7e15:9756:5c67:e283:16e]:3000
2021/09/20 02:21:42.079534 [info] Go to http://172.24.0.1:3000
2021/09/20 02:21:42.079536 [info] Go to http://172.23.0.1:3000
2021/09/20 02:21:42.079537 [info] Go to http://172.20.0.1:3000
2021/09/20 02:21:42.079539 [info] Go to http://172.17.0.1:3000
2021/09/20 02:21:42.079540 [info] Go to http://172.22.0.1:3000
2021/09/20 02:21:42.079541 [info] Go to http://172.29.0.1:3000
2021/09/20 02:21:42.079543 [info] Go to http://192.168.0.1:3000
2021/09/20 02:21:42.079545 [info] Go to http://172.27.0.1:3000
2021/09/20 02:21:42.079546 [info] Go to http://172.25.0.1:3000
2021/09/20 02:21:42.079547 [info] Go to http://172.31.0.1:3000
2021/09/20 02:21:42.079549 [info] Go to http://172.26.0.1:3000
2021/09/20 02:21:42.079550 [info] Go to http://172.28.0.1:3000
2021/09/20 02:21:42.079552 [info] Go to http://172.21.0.1:3000
2021/09/20 02:21:42.079553 [info] Go to http://172.18.0.1:3000
2021/09/20 02:21:42.079555 [info] Go to http://172.19.0.1:3000
2021/09/20 02:21:42.079556 [info] Go to http://172.16.238.1:3000
2021/09/20 02:21:42.079558 [info] Go to http://192.168.203.1:3000
2021/09/20 02:23:28.367187 [info] Starting the DNS proxy server
2021/09/20 02:23:28.367217 [info] Ratelimit is enabled and set to 20 rps
2021/09/20 02:23:28.367224 [info] The server is configured to refuse ANY requests
2021/09/20 02:23:28.367230 [info] DNS cache is enabled
2021/09/20 02:23:28.367238 [info] MaxGoroutines is set to 300
2021/09/20 02:23:28.367249 [info] Creating the UDP server socket
2021/09/20 02:23:28.367371 [info] Listening to udp://[::]:53
2021/09/20 02:23:28.367385 [info] Creating a TCP server socket
2021/09/20 02:23:28.367448 [info] Listening to tcp://[::]:53
2021/09/20 02:23:28.367570 [info] Entering the UDP listener loop on [::]:53
2021/09/20 02:23:28.367700 [info] Entering the tcp listener loop on [::]:53
2021/09/20 02:23:28.491656 [info] Go to http://192.168.50.138:8662
```

**Reply by bigverm23** [reply]
yep, network_mode: host worked for me.  I find it interesting I couldn't put the container directly onto a reverse proxy network using the compose.
